ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/displayedgraph/ProcessStatus.scala

```diff
@@ -33,6 +32,27 @@ object ProcessStatus {
   def apply(status: StateStatus, processStateDefinitionManager: ProcessStateDefinitionManager): ProcessStatus =
     ProcessStatus(status, processStateDefinitionManager, Option.empty, Option.empty, Option.empty, List.empty)
 
+  def error(status: ErrorStateStatus,
+            processStateDefinitionManager: ProcessStateDefinitionManager,
+            user: String,
+            versionId: Long,
+            processState: Option[ProcessState] = Option.empty,
+            exceptedVersion: Option[Long] = Option.empty): ProcessStatus = {
+    ProcessStatus(
+      status,
+      processState.map(_.deploymentId.value),
+      allowedActions = processStateDefinitionManager.statusActions(status),
+      icon = processStateDefinitionManager.statusIcon(status),
+      tooltip = processStateDefinitionManager.statusTooltip(status).map(_.format(
```

Comment:
What will happen if someone will change the order of placeholders in tooltip? How developer would know what is the correct order? How developer will find out that order of variables was changed in some versions in a future?
You have status classes, you can put there useful info like version, you can use this version during tooltip preparing. You just need to make Simple*Manager more OO friendly - use methods instead of maps...

Reply:
I agree. But there is also one more problem: we mix statuses from ProcessManager and ManagementActor. Here we assume that processStateDefinitionManager has to handle (some) statuses from SimpleProcessManager?

Reply:
We have TODO's that at the end we want to hide `ManagementActor` behind ProcessManager's API. But indeed, during this transition we can handle this situation better. Two ideas that I have:
1. Maybe we should pass to tooltip method some additional "deployment details" which will be used during formatting
2. Maybe we can extract "base statuses" from "simple statuses". In fact this cases are quite specific for situation when you have "remote engine" that can be desynchronized. Let's assume local engine (like in NiFi). In this scenario you won't have such statuses. Maybe we should call those extracted statuses "remote statuses"?